### DISCUSSION TITLE
sendEmail Activity Elsa3.0 smtp config

### DISCUSSION THREAD

**Discussion by EK591892** [elsa-workflows/elsa-core#5552] [Q&A]
Hello, 
I'm exploring the ELSA workflow activities and i'm questioning on how to set the SMTP config for the SendEmail activity to work. I'm not using any application to host server or studio. Simply, loaded an image (Server and Studio) in docker. The container responds on http://localhost:13000. my workflow is basic, Httpendpoint then sendemail. The container responds to postman calls with no error but no email is sent.

Thanks,
![Elsa basic workflow](https://github.com/elsa-workflows/elsa-core/assets/55601698/ecf837f0-86c3-48d1-8c8f-513cf74ad011)

**Comment by glime-ai** [comment]
To configure the SMTP settings for the SendEmail activity in Elsa when using Docker, you will typically need to set environment variables for the Docker container that specify your SMTP server details. These details include the SMTP host, port, username, password, and other related settings.

Since you are using the Docker image for Elsa Server and Studio, you can pass the SMTP configuration as environment variables when you start the Docker container. Here's an example of how you might set these variables:

```shell
docker run -e SMTP__Host=smtp.example.com -e SMTP__Port=587 -e SMTP__Username=your-username -e SMTP__Password=your-password -e SMTP__EnableSsl=true -p 13000:80 elsaworkflows/elsa-server-and-studio-v3:latest
```

Replace `smtp.example.com`, `587`, `your-username`, and `your-password` with your actual SMTP server details. The `SMTP__EnableSsl` variable should be set to `true` if your SMTP server requires SSL.

Please note that the environment variable names are case-sensitive and should be prefixed with `SMTP__` followed by the setting name.

After setting these environment variables, the SendEmail activity should be able to send emails using the specified SMTP server.

If you need to make changes to the SMTP settings after the container is already running, you would need to restart the container with the updated environment variables.

Remember to ensure that the SMTP server you are using allows connections from the host where your Docker container is running, and that any firewalls or security groups are configured to allow outbound email traffic on the SMTP port you are using.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by EK591892** [comment]
Thank you for this quick reply. I have tried to run the command you proposed with a different value such smtp__host =locahost and smtp__port=2525 which is a running container (smtp4dev)

I tried even copying the docker run command from the existing container to run it in CMD and i' hitting other errors such the reference name should be in lowercase.
________________________________
